### PR TITLE
Change default kubelet IP and pod IP range

### PR DESCRIPTION
**Pull request by ZakariasL**
Seems the default IP 10.0.0.50 will overlap easily with routers that use the 10.0.0.x address space as the default address space.

A customer ran into this problem and thus we should avoid using these addresses. Instead, we use `172.21.1.0/30` for Node IP and  `172.21.2.0/24` Pod CIDR. Additionally:
- A script is added to modify these addresses on the gateway
- The subnet of the kubelet is reduced from 24->30 
- We increase the kaas0 interface metric to 999, reducing the priority in case of conflicts with local networks

A more permanent solution is to automatically detect conflicts and dynamically update the Node and Pod IPs. This is a temporary solution until automatic IP conflict resolution is implemented.

Before deciding on this solution, we discussed using link-local addresses [in this PR](https://github.com/PelionIoT/meta-pelion-edge/pull/343)
Internal ref: https://armiot.atlassian.net/browse/IOTNGGW-261

### CODE REVIEW

**Review by JanneKiiskila — DISMISSED** [review]
LGTM

File: recipes-containers/kubelet/files/set-kubelet-ip.sh [hunk 1]
```diff
@@ -1,10 +1,42 @@
 #!/bin/bash
+# ----------------------------------------------------------------------------
+# Copyright (c) 2021, Pelion and affiliates.
+# Copyright (c) 2020, Arm Limited and affiliates.
```
Comment (JanneKiiskila) [inline recipes-containers/kubelet/files/set-kubelet-ip.sh:4]: We can drop this line.

File: recipes-containers/kubelet/files/set-kubelet-ip.sh [hunk 2]
```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+# ----------------------------------------------------------------------------
+# Copyright (c) 2021, Pelion and affiliates.
+# Copyright (c) 2020, Arm Limited and affiliates.
```
Comment (x85446) [inline recipes-containers/kubelet/files/set-kubelet-ip.sh:4]:
```suggestion
```

File: recipes-containers/kubelet/files/set-kubelet-ip.sh (hunk 1 above)
Comment (x85446) [reply recipes-containers/kubelet/files/set-kubelet-ip.sh:4]: suggestion in place

**Review by x85446 — DISMISSED** [review]
LGTM

File: recipes-containers/kubelet/files/set-kubelet-ip.sh (hunk 1 above)
Comment (ZakariasL) [reply recipes-containers/kubelet/files/set-kubelet-ip.sh]: Done.

**Comment by ZakariasL** [comment]
Validated on an RPI3B+ board, ready to be merged. @JanneKiiskila @x85446

File: recipes-containers/kubelet/files/set-kubelet-ip.sh [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+# Change the Pelion kubelet IP and Pod CIDR.
+#
+# 1. NODE_IP sets the kubelet IP by modifying the `kaas0` interface.
+# 2. POD_CIDR and POD_CIDR_GW sets the IP range for Pods. The POD_CIDR_GW must 
+#    correspond to the gateway of the interface created from the POD_CIDR 
+#    subnet. For /24 subnet, this would normally be the IP with the last number 
+#    set to "1".
+# 3. Run the script.
+
+NODE_IP="172.21.1.0"
```
Comment (JanneKiiskila) [inline recipes-containers/kubelet/files/set-kubelet-ip.sh:30]: The idea of modifying these on the file might be a tad bit difficult, as you might not have rights to modify this file (if it's part of the OSTree managed area, it's in read only mode).  Ideally the setting script should take these as parameters on the command line instead @ZakariasL ?

@kivaisan  @petedyerarm - how do you feel?

Comment (JanneKiiskila) [reply recipes-containers/kubelet/files/set-kubelet-ip.sh:30]: I guess short term workaround is to advice people to copy this over to some place, where they CAN modify it and then run it. We need a docs topic to pelion-dm-edge-docs. @ZakariasL  @AnotherButler

Comment (ZakariasL) [reply recipes-containers/kubelet/files/set-kubelet-ip.sh:30]: Ah, I assumed it was read/write mode for everyone based on how it was copied to the location: https://github.com/PelionIoT/meta-pelion-edge/blob/74c8fb2bab93b6c622895adb0b5c5210932885bb/recipes-containers/kubelet/kubelet_git.bb#L62

Comment (ZakariasL) [reply recipes-containers/kubelet/files/set-kubelet-ip.sh:30]: My bad, I see now that it only has write permission for the owner of the file. Let me know if I should make it into arguments instead

**Review by x85446 — APPROVED** [review]
Temporary for 2.5.  Will go to autodetect in 2.6

**Pull request merged**